include/inviwo/core/ports/outport.h

```diff
@@ -60,12 +63,16 @@ class IVW_CORE_API Outport : public Port {
     /**
      * An outport is ready if it has data and is valid. The port is set valid after its
      * processor successfully finishes processing.
+     * @pre isReady_ must be set by derived the class.
      * @see invalidate @see setValid
      */
     virtual bool isReady() const override;
 
     /**
      * Called by Processor::invalidate, will invalidate its connected inports.
+     * @pre isReady_ must be set by the derived class.
```

Comment:
How is isReady relevant for the invalidation/setValid? They have very little to do with each other I think

Reply:
isReady_ is called on invalidate
```
void Outport::invalidate(InvalidationLevel invalidationLevel) {
    invalidationLevel_ = invalidationLevel;
    for (auto inport : connectedInports_) inport->invalidate(invalidationLevel);
    isReady_.update();
}
```

Reply:
true, but we call isReady_.update() in most non-const functions. I feel like that information does not belong in the documentation of each public member. It is an implementation detail. And only people that derive from outport needs to know about it. Not users of ports in general.

Reply:
Sure, probably enough with the note in the class documentation